### PR TITLE
feat(leaves): Factor out cursor as a leaf

### PR DESCRIPTION
**Pull request by nguermond**
feat(leaves): 
- Factor out cursor as a leaf (as in bubble)
- allow blinking cursor
- add set_text to text_input

### CODE REVIEW

**Comment by leostera** [comment]
Thanks for the contrib @nguermond! ✨ – could you rerecord the text input demo gif? It'll make it easier to see how this behaves.

You can run `vhs demo.tape` from the example folder. Just make sure to have the latest VHS installed: https://github.com/charmbracelet/vhs/?tab=readme-ov-file#installation

**Comment by nguermond** [comment]
I've updated the example with the gif. Let me know if there are other issues.

File: leaves/cursor.ml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+let make
+      ?(style=Spices.(default |> bg white |> fg black))
+      ?(blink=true)
+      ?(fps=Fps.of_float 2.5) ()
+  = {
+    focus = true;
+    blink = blink;
+    fps = fps;
+    show = true || (not blink);
```
Comment (leostera) [inline leaves/cursor.ml:27]: should this be `not blink`?

File: leaves/cursor.ml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+let update t (e : Minttea.Event.t) =
+  match e with
+  | Frame now when t.blink ->
+     if (Fps.tick ~now t.fps) = `frame then
+       let show = not t.show in
+       { t with show }
+     else t
+  | _ -> t
+
+let view t ?(text_style=Spices.(default |> bg black |> fg white)) str =
```
Comment (leostera) [inline leaves/cursor.ml:41]: Can we put the default text style in a value? `default_text_style` would be ok 👌🏼

**Comment by leostera** [comment]
Left you some small changes, after that I think it's good to merge. Make sure to run `dune fmt` too please 🙏🏼 

Thanks so much for the contribution ✨

File: leaves/cursor.ml (hunk 1 above)
Comment (nguermond) [reply leaves/cursor.ml:27]: This should just be `true`, thanks for catching this!

File: leaves/cursor.ml (hunk 2 above)
Comment (nguermond) [reply leaves/cursor.ml:41]: `Spices.default` should be sufficient here actually

**Comment by nguermond** [comment]
Should be fixed now!

**Comment by leostera** [comment]
Amazing! Thank you so much ✨

**Pull request merged**